### ISSUE TITLE
Error: Struct() argument 1 must be string, not unicode using mesh.Mesh.from_file() 

### ISSUE THREAD

**Issue by futureneer** [WoLpH/numpy-stl#10]
Hello,
I get the following error using the command line functions:

```bash
$ stl2ascii right_hand_pointing.stl right_hand_pointing_bin.stl  
Traceback (most recent call last):
  File "/usr/local/bin/stl2ascii", line 9, in <module>
    load_entry_point('numpy-stl==1.5.0', 'console_scripts', 'stl2ascii')()
  File "/usr/local/lib/python2.7/dist-packages/stl/main.py", line 68, in to_ascii
    remove_empty_areas=args.remove_empty_areas)
  File "/usr/local/lib/python2.7/dist-packages/stl/stl.py", line 270, in from_file
    name, data = cls.load(fh, mode=mode)
  File "/usr/local/lib/python2.7/dist-packages/stl/stl.py", line 70, in load
    data = cls._load_binary(fh, header, check_size=True)
  File "/usr/local/lib/python2.7/dist-packages/stl/stl.py", line 79, in _load_binary
    count, = struct.unpack('@i', b(fh.read(COUNT_SIZE)))
TypeError: Struct() argument 1 must be string, not unicode
```

I also get this problem when using the api, as in the example:

```python
your_mesh = mesh.Mesh.from_file('right_hand_pointing.stl')
```

I am using the newest version of python-utils, with python 2.7.3. Is this correct?

Thanks!
Kel

**Comment by futureneer** [comment]
Also, I am pretty sure that I am sending a string, because of the following output from ipython with no imports:

```python
In [4]: type('a')
Out[4]: str
```

**Comment by futureneer** [comment]
Ok, so I managed to fix this by changing line 79 in stl.py from
```python
count, = struct.unpack('@i', b(fh.read(COUNT_SIZE)))
```
to 
```python
count, = struct.unpack(str('@i'), b(fh.read(COUNT_SIZE)))
```

Again, I am not sure if this is an issue with my system configuration or this code. I have seen this fix elsewhere to address this error.

Thanks!
Kel

**Comment by WoLpH** [comment]
Hi Kel,

The problem here is your file object, especially with Python 3 this can return either strings or bytes, which make things difficult... on Python 2 I haven't had any issues anymore so I'm thinking it's related to one of the installed libraries.

Do you have pdbpp installed by any chance? That breaks the code for me.

**Comment by loicgasser** [comment]
I am using python 2.7.3 and I observed the same problem.
I know where it comes from. On top of your stl.py lib you have

```python
from __future__ import unicode_literals
```
This will automatically convert all strings to unicodes. In python 3 all strings are unicode which is not the case in python 2.7.x.

struct expect a string as first arg, but it only receives a unicode.

Try:

```python
[GCC 4.7.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> myval='@i'
>>> type(myval)
<type 'str'>
```

and then

```python
>>> from __future__ import unicode_literals
>>> myval='@i'
>>> type(myval)
<type 'unicode'>
```

I can provide a fix for that if you want.

**Comment by loicgasser** [comment]
Ah just saw you fixed that in https://github.com/WoLpH/numpy-stl/commit/2ae619377286f0b6db7cdc32b16cdb6e486cd9af but this is not yet published on pypi.

**Comment by WoLpH** [comment]
This was actually fixed a very long time ago (version 1.5.1, we're at 2.0.0 now). Which version are you running?

**Comment by loicgasser** [comment]
2.0.0 and it's not there it seems https://github.com/WoLpH/numpy-stl/blob/v2.0.0/stl/stl.py#L287
